ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/QueryResponseHandler.java

```diff
@@ -35,10 +52,32 @@
 
   @Override
   protected void doHandleBodyBuffer(final Buffer buff) {
+    final Buffer strippedValidJson = toJsonMsg(buff, true);
+
+    if (strippedValidJson.toString().equals("")) {
+      return;
+    }
+
     if (!hasReadArguments) {
-      handleArgs(buff);
+      handleArgs(strippedValidJson);
     } else {
-      handleRow(buff);
+      final JsonNode rowNode;
+      try {
+        rowNode = JSON_MAPPER.readTree(strippedValidJson.getBytes());
+      } catch (Exception e) {
+        cf.completeExceptionally(e);
+        return;
+      }
+
+      if (rowNode.has(FINAL_MESSAGE_KEY)) {
+        doHandleBodyEnd();
```

Comment:
This is now called twice.  I'm not sure that's an issue, but we may want to avoid that.  Maybe just log a message instead?

Reply:
done